packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/platform_interface/google_maps_inspector_platform.dart

```diff
@@ -0,0 +1,117 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:google_maps_flutter_platform_interface/google_maps_flutter_platform_interface.dart';
+import 'package:plugin_platform_interface/plugin_platform_interface.dart';
+
+/// The interface that platform-specific implementations of
+/// `google_maps_flutter` can extend to support state inpsection in tests.
+///
+/// Avoid `implements` of this interface. Using `implements` makes adding any
+/// new methods here a breaking change for end users of your platform!
+///
+/// Do `extends GoogleMapsInspectorPlatform` instead, so new methods
+/// added here are inherited in your code with the default implementation (that
+/// throws at runtime), rather than breaking your users at compile time.
+abstract class GoogleMapsInspectorPlatform extends PlatformInterface {
+  /// Constructs a GoogleMapsFlutterPlatform.
+  GoogleMapsInspectorPlatform() : super(token: _token);
```

Comment:
I don't think a base `GoogleMapsFlutterPlatform` would likely to be useful to an implementation; it would need to be a specific implementation with extra accessors. So if we take it and store it here, it would have to be downcast all the time, whereas if we take and store it in the implementations, it can already be the correct type for each implementation.

Also, we could pretty easily have an inspector that didn't take a full instance. If the method channel implementation weren't already exposing the channel mapping, I would make its inspector take a function that returns the channel for a given ID since that's the only thing we need. I may well do that when I split out platform-specific Dart implementations for Android and iOS.

Reply:
Ah yes, having to cast whatever is passed around sounds ugly :/